File: programs/ln007/rect.py
class Rect:
   def __init__(self, x, y):
      self.xdim = x
      self.ydim = y

   def area (self):
      return self.xdim * self.ydim  # accessing internal identity via self

r = Rect(2.0, 4.0)
a = r.area ()
assert (a == 8.0)

File: programs/ln007/rect.rs
struct Rectangle {
    xdim: f32,
    ydim: f32,
}

impl Rectangle {
    fn area(&self) -> f32 {
        self.xdim * self.ydim
    }
}

fn main() {
    let r = Rectangle { xdim: 2.0, ydim: 4.0 };
    let a = r.area();
    assert!(a == 8.0);
}
